java/src/main/java/ai/rapids/cudf/GroupByResult.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.rapids.cudf;
+
+/**
+ * Used by JNI
+ * Used to save groups and keys for `contiguousSplitGroupsAndGenUniqKeys`
+ * Each row in uniq keys table is corresponding to a group
+ */
+public class GroupByResult {
```

Comment:
I am not very happy with how the resources an managed here. This container can hold a large number of different objects that need to be closed, and the only way to close them is to pull the objects out of it and close them each separately.  Could we make this AutoClosable and have it close all of the tables/ContiguousTables inside of it? Because Tables themselves are not reference counted it might be good to have a `releaseGroups` API in addition to the getGroups API so the user can pull the groups out of the GroupByResult and take ownership of them.

Reply:
Done